msgcntr/messageforums-component-impl/src/test/java/org/sakaiproject/component/app/messageforums/MsgcntrTestConfiguration.java

```diff
@@ -0,0 +1,359 @@
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.springframework.orm.hibernate.AdditionalHibernateMappings;
+import org.sakaiproject.tasks.api.TaskService;
+import org.sakaiproject.thread_local.api.ThreadLocalManager;
+import org.sakaiproject.time.api.TimeService;
+import org.sakaiproject.time.api.UserTimeService;
+import org.sakaiproject.tool.api.ActiveToolManager;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolManager;
+import org.sakaiproject.user.api.PreferencesService;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotificationPreferencesRegistration;
+import org.sakaiproject.util.api.FormattedText;
+import org.sakaiproject.util.api.LinkMigrationHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.orm.hibernate5.HibernateTransactionManager;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBuilder;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import javax.annotation.Resource;
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.util.Properties;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Created by chmaurer on 11/29/20.
+ */
+@Configuration
+@EnableTransactionManagement
+@ImportResource({"classpath:/WEB-INF/components.xml", "classpath:/test-components.xml"})
+@PropertySource("classpath:/hibernate.properties")
+public class MsgcntrTestConfiguration {
+
+    @Autowired
+    private Environment environment;
+
+    @Resource(name = "org.sakaiproject.springframework.orm.hibernate.impl.AdditionalHibernateMappingsImpl.messageforum")
+    private AdditionalHibernateMappings hibernateMappings;
+
+    static {
+        System.setProperty("auto.ddl", "true");
+    }
+
+    @Bean(name = "org.sakaiproject.springframework.orm.hibernate.GlobalSessionFactory")
+    public SessionFactory sessionFactory() throws IOException {
+        DataSource dataSource = dataSource();
+        LocalSessionFactoryBuilder sfb = new LocalSessionFactoryBuilder(dataSource);
+        StandardServiceRegistryBuilder srb = sfb.getStandardServiceRegistryBuilder();
+        srb.applySetting(org.hibernate.cfg.Environment.DATASOURCE, dataSource);
+        srb.applySettings(hibernateProperties());
+        StandardServiceRegistry sr = srb.build();
+        sr.getService(MutableIdentifierGeneratorFactoryInitiator.INSTANCE.getServiceInitiated())
+                .register("uuid2", AssignableUUIDGenerator.class);
```

Comment:
Is this needed? If not can ellminate the srb pieces.

Reply:
Not sure.  I just followed an example from somewhere else!  I'll take a look...

Reply:
Was able to remove some stuff!